moseq2_model/helpers/wrappers.py

```diff
@@ -170,6 +171,59 @@ def learn_model_wrapper(input_file, dest_file, config_data):
         return img_path
 
 
+def apply_model_wrapper(model_file, pc_file, dest_file, config_data):
+    """
+    Wrapper function to apply a pre-trained model to new data.
+
+    Args:
+    model_file (str): Path to pre-trained model file
+    pc_file (str): Path to PC scores file
+    dest_file (str): Path to save output file
+
+    Returns:
+    None
+    """
+
+    assert splitext(basename(dest_file))[-1] in ['.mat', '.z', '.pkl', '.p', '.h5'], 'Incorrect model filetype'
+    os.makedirs(dirname(dest_file), exist_ok=True)
+
+    if not os.access(dirname(dest_file), os.W_OK):
+        raise IOError('Output directory is not writable.')
+
+
+    # Load model
+    model_data = load_dict(model_file)
+
+    # Load PC scores
+    data_dict, data_metadata = load_pcs(filename=pc_file, var_name=config_data.get('var_name', 'scores'), npcs=model_data['run_parameters']['npcs'],
+                                        load_groups=config_data.get('load_groups', False))
+
+    # parse group information from index file
+    index_data, data_metadata = process_indexfile(config_data.get('index', None), data_metadata,
+                                                  config_data.get('default_group', 'n/a'), select_groups=False)      
+
+    # Apply model
+    syllables = apply_model(model_data['model'], model_data['whitening_parameters'], data_dict, data_metadata, model_data['run_parameters']['whiten'])
+
+    # add -5 padding to the list of states
+    nlags = model_data['run_parameters'].get('nlags', 3)
+    for key in syllables.keys():
+        syllables[key] = np.append(np.repeat(-5, nlags), syllables[key])
+
+    # prepare model data dictionary to save
+    applied_model_data = {}
+    applied_model_data['labels'] = list(syllables.values())
+    applied_model_data['keys'] = list(syllables.keys())
+    applied_model_data['model_parameters'] = model_data['model_parameters']
+    applied_model_data['oracle_run_parameters'] = model_data['run_parameters']
```

Comment:
I originally added `model_path` key but then decided not to because the model parameters are all saved in the newly applied model object, and it is possible that users don't keep the old model at the same model path after they applied the model to new data so I decided to remove this field. As for `pc_path` if you are talking about the pc_file with all the PC components the new data is projected to, the information is saved in the `config.yaml`, consistent with the original freshly trained models.

Reply:
134aef52109fe30e048fbe30b6f266b2f2b85d59 the for loop is in this commit.